scripts/deleteAndLoadSnapsets.js

```diff
@@ -0,0 +1,398 @@
+    process.exit(1);
+};
+
+/**
+ * General mechanism to create a database request, set up an error handler and
+ * return.  It is up to the caller to trigger the request by calling its end()
+ * function.
+ * @param {String} databaseURL - URL to query the database with.
+ * @param {Function} handleResponse - callback that processes the response from
+ *                                    the request.
+ * @param {String} errorMsg - optional error message for request errors.
+ * @return {http.ClientRequest} - The http request object.
+ */
+dbLoader.queryDatabase = function (databaseURL, handleResponse, errorMsg) {
+    var aRequest = http.request(databaseURL, handleResponse);
+    aRequest.on("error", function (e) {
+        fluid.log(errorMsg + e.message);
+    });
+    return aRequest;
+};
+
+/**
+ * Get all the json files from the given directory, then loop to put their
+ * contents into an array of Objects.
+ * @param {String} dataDir - Directory containing the files to load.
+ * @return {Array} - Each element of the array is an Object based on the
+ *                   contents of each file loaded.
+ */
+dbLoader.getDataFromDirectory = function (dataDir) {
+    var contentArray = [];
+    var files = fs.readdirSync(dataDir);
+    files.forEach(function (aFile) {
+        if (aFile.endsWith(".json")) {
+            var fileContent = fs.readFileSync(dataDir + "/" + aFile, "utf-8");
+            contentArray = contentArray.concat(JSON.parse(fileContent));
+        }
+    });
+    return contentArray;
+};
+
+/*
+ * Create the step that fetches the current snapset Prefs Safes from the
+ * database.
+ * @param {Object} options - Object that has the view for finding snap set
+ *                           Prefs Safes records in the database.
+ * @return {Object} - An object containing the database request to load the
+ *                    static data, and a promise configure to trigger the next
+ *                    step.
+ */
+dbLoader.createFetchSnapsetsStep = function (options) {
+    var togo = fluid.promise();
+    var response = dbLoader.createResponseHandler(
+        dbLoader.processSnapsets,
+        togo,
+        "Error retrieving snapsets Prefs Safes: "
+    );
+    var snapsSetsRequest = dbLoader.queryDatabase(
+        options.prefsSafesViewUrl,
+        response,
+        "Error requesting snapsets Prefs Safes: "
+    );
+    return { request: snapsSetsRequest, promise: togo };
+};
+
+/*
+ * Create the step that fetches the current GPII keys associate with the snapset
+ * Prefs Safes.
+ * @param {Object} options - Object that has the view for finding all GPII Key
+ *                           records in the database.
+ * @param {Promise} previousStep - Promise from a previous step whose fulfillment
+ *                                 triggers the request configured herein.
+ * @return {Promise} - The promise asoociated with this step.
+ */
+dbLoader.createFetchGpiiKeysStep = function (options, previousStep) {
```

Comment:
Eliminate the "previousStep" pattern from all of these, and put this responsibility onto the caller to orchestrate.